DRAFT: Slider Label and Value feature

lenadax opened the pull request:
Currently used Version: 5.5.2

Description
========

This is a DRAFT for a feature that implements a label and an editable input field for the iro.js Slider component.
As an unplanned 'side effect', I also added an option to manually specify the slider length!

Seeing as it's my first take on typescript and preact, I am sure there is quite a few things to modify, but I would be more than happy to hear about your thoughts on implementing this feature and the basic structure in this PR.

A note: I didn't take vertical sliders into full consideration when creating this branch. It works, but it's not too pretty for now.
I also did not update or write any new tests yet, I would be happy to hear your opinion first.

See the corresponding repo for iro/core: https://github.com/lenadax/iro-core/tree/feature/slider-label-value


Motivation
========

I found that, having more than a few sliders, it really wasn't clear anymore which slider was changing which value, so I decided to fork and try to solve the issue, potentially increasing clarity for other users as well.
A project I am working on also requires channel values to be displayed for each slider, and editing single channels as well.

I think having the option to display each channel's value is a neat addition to the existing functionality, especially since the nature of iro's build makes it hard to add this kind of content externally.


Changes
=======

Modified: slider.tsx
-------------------------
Slider props now accept two now options: showInput and disabled (both type boolean).
The showInput property renders an input field and label when true, the disabled property disables the input field for user interaction.
In order to allow for more contained positioning, I allowed the slider width to be different from box width, so by default slider and input combined fit neatly under the iro Box. For positioning, I added a wrapper around the slider elements and set display to flex:none instead of block:none.
In order to set a slider's length (by default, width), one can pass a sliderLength property, but enabling input field overrides this behaviour.
  
New file: input.tsx
-----------------------
The Input component renders a span and input field for each slider, updating its value based on activeColor.
On user interaction (typing or copying from clipboard), the activeColor value gets updated as well.
I used the clamp() function coupled with Regex in iro/core to validate user input.

jaames commented:
Hey Lena! Sorry for not checking in here sooner, just wanted to say this is a really cool feature idea and everything looks pretty solid so far. Please let me know if there's anything I can do to support this effort :)

As for tests, I honestly wouldn't worry about that for now. I'm hoping to completely redo how I do tests (and fully rewrite everything) with iro.js 6.0 whenever I get around to it.

lenadax commented:
Hi James! No Problem at all, thank you for replying!

I added and switched up a few more little things in the PR, so everything should run fairly smoothly now.
From my side, I think this feature would be considered complete. It would be nice if you could find the time to look over the code a bit, and hopefully merge if it fits your requirements :)
If there is any problems regarding the added feature, I'd be happy to modify things as needed.

rnixx commented:
Hi @jaames,

any news on this one?

jepsar commented:
I would love to see this in the next release, as this will be great a accessibility improvement as well. See also #237 and #238.

mydudechris commented:
Any news on adding this to the next release?

jaames commented:
As mentioned at the top of the readme; the project is currently on hiatus until I have the time/money to work on it again. Thanks.
